File: EXAMEN2/app/src/main/java/com/example/examen2/MainActivity.java
package com.example.examen2;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.graphics.Path;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    ObjectAnimator animation1, animation2, animation3;

    ObjectAnimator[] animations;
    ImageView mongolfiere;
    Button start;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        animations = new ObjectAnimator[6];

        Path p = new Path();

        p.lineTo(800, 500);
        Path p2 =  new Path();
        p2.lineTo(500, 300);
        Path p3 =  new Path();
        p3.lineTo(650, 100);
        mongolfiere = findViewById(R.id.mongolfiere);
        start = findViewById(R.id.buttonStart);

        animations[0] = ObjectAnimator.ofFloat(mongolfiere, ImageView.X, 800);
        animations[0].setDuration(2000);

        animations[1] = ObjectAnimator.ofFloat(mongolfiere, ImageView.Y, 900);
        animations[1].setDuration(2000);

        animations[2] = ObjectAnimator.ofFloat(mongolfiere, ImageView.X, 400);
        animations[2].setDuration(2000);

        animations[3] = ObjectAnimator.ofFloat(mongolfiere, ImageView.Y, 300);
        animations[3].setDuration(2000);

        animations[4] = ObjectAnimator.ofFloat(mongolfiere, ImageView.X, 400);
        animations[4].setDuration(2000);

        animations[5] = ObjectAnimator.ofFloat(mongolfiere, ImageView.Y, 200);
        animations[5].setDuration(2000);

        AnimatorSet set = new AnimatorSet();
        set.playTogether(animations[0], animations[1]);
        AnimatorSet set2 = new AnimatorSet();
        set.playTogether(animations[2], animations[3]);
        AnimatorSet set3 = new AnimatorSet();
        set3.playTogether(animations[4], animations[5]);
//
//        animation1 = ObjectAnimator.ofFloat(mongolfiere, ImageView.X, 800);
//        animation2 = ObjectAnimator.ofFloat(mongolfiere, ImageView.Y, 200);
//
//        AnimatorSet set = new AnimatorSet();
//        set.setDuration(2000);
//        set.playTogether(animation1, animation2);
//
            for (int i = 0; i < animations.length; i++){
                if (i == 0) {
                    start.setOnClickListener(v -> set.start());
                }
                else if (i == 2){
                        start.setOnClickListener(v -> set2.start());
                }
                else if (i == 4) {
                    start.setOnClickListener(v -> set3.start());
                }
            }

    }
}

File: examen1_kt/app/src/main/java/com/example/examen1_kt/MainActivity.kt
package com.example.examen1_kt

import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.BufferedReader
import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.InputStreamReader
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.util.Date
import java.util.Vector

class MainActivity : AppCompatActivity() {
    lateinit var langages: Vector<Langages>
    lateinit var nbChamp: TextView
    lateinit var classement2012:TextView
    lateinit var texteDate:TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        langages = Vector()
        texteDate = findViewById(R.id.texteDate)

        try {
            val derniereFois = deserialisation()
            val formatter = DateTimeFormatter.ofPattern("hh:mm dd MM yyyy")
            texteDate.setText("derniere utilisation: " + derniereFois.format(formatter))
        } catch (e: FileNotFoundException){
            texteDate.setText("1ere utilisation")
        }

        nbChamp = findViewById(R.id.champClass2012)
        classement2012 = findViewById(R.id.Class2012)


        lectureFichierLangage()

        nbChamp.setText(trouver99())




    }


    override fun onStop() {
        super.onStop()
        serialisation()
    }
    fun lectureFichierLangage() {



        val fis: FileInputStream = openFileInput("langage.txt")
        val obj = InputStreamReader(fis)
        val br = BufferedReader(obj)
        br.readLine()
        fis.bufferedReader().use {
            var line = br.readLine()

            while (line != null) {
                var tab = line.split(",")
                langages.add(Langages(tab[0], tab[1].toInt(), tab[2].toInt(), tab[3].toInt()))
                line = br.readLine()
            }
        }
    }

    fun trouver99():Int{
        var compteur:Int = 0
        for(l:Langages in langages){
            if (l.classement2012 == 99)
                compteur++
        }
        return compteur
    }

    fun serialisation(){

        val fos = openFileOutput("date.ser", MODE_PRIVATE)
        val oos = ObjectOutputStream(fos)

        oos.use {oos ->
            oos.writeObject(LocalDateTime.now())
        }
    }

    fun deserialisation(): LocalDateTime {
        openFileInput("date.ser").use { fos ->
            ObjectInputStream(fos).use { oos ->
                return oos.readObject() as LocalDateTime
            }
        }
    }
}